lib/active_model/serializer/fieldset.rb

```diff
@@ -18,14 +18,23 @@ def fields_for(type)
       attr_reader :raw_fields
 
       private
-
       def parsed_fields
+        # TODO: this is not very flexible.
+        # - we should probably support symbol, array, hash, etc
+        #
+        # JSONAPI::IncludeDirective supports these, but also changes the resulting structure.
+        # is it worth it to duplicate some of that functionality?
+        # or should IncludeDirective be modified to only allow one layer of options?
```

Comment:
If we can return to a clean adapterless cache unaware serializer that can take in options to serialize its attributes and association... and I'd love to use just functions here, like in knuckles, we could rock this